src/signer/rfqv1.ts

```diff
@@ -29,25 +31,75 @@ export async function signByEOA(orderHash: string, wallet: Wallet): Promise<stri
   return signature
 }
 
-// For V4 Maket Maker Proxy (MMP)
-// Signature:
-// +------|---------|---------|---------|---------|---------+
-// |  V   |    R    |    S    |userAddr |feeFactor| type(6) |
-// +------|---------|---------|---------|---------|---------+
 export function signByMMPSigner(
   orderHash: string,
   userAddr: string,
   feeFactor: number,
   wallet: Wallet,
-  signatureType: SignatureType
+  walletType: WalletType
 ): string {
-  let signature = signWithUserAndFee(wallet, orderHash, userAddr, feeFactor)
-  const signatureBuffer = Buffer.concat([
-    ethUtils.toBuffer(signature),
-    ethUtils.toBuffer(signatureType),
-  ])
-  signature = '0x' + signatureBuffer.toString('hex')
-  return signature
+  if (walletType === WalletType.MMP_VERSOIN_4) {
+    // For V4 Maket Maker Proxy (MMP)
+    // Signature:
+    // +------|---------|---------|---------|---------|---------+
+    // |  V   |    R    |    S    |userAddr |feeFactor| type(6) |
+    // +------|---------|---------|---------|---------|---------+
+    let signature = signWithUserAndFee(wallet, orderHash, userAddr, feeFactor)
+    const signatureBuffer = Buffer.concat([
+      ethUtils.toBuffer(signature),
+      ethUtils.toBuffer(SignatureType.Wallet),
+    ])
+    signature = '0x' + signatureBuffer.toString('hex')
+    return signature
+  } else if (walletType === WalletType.MMP_VERSION_5) {
+    // |1 byte| 32 byte | 32 byte | 1 byte  |
+    // +------|---------|---------|---------+
+    // |  V   |    R    |    S    | type(6) |
+    // +------|---------|---------|---------+
+    let signature = ecSignOrderHash(
+      wallet.privateKey.slice(2),
+      orderHash,
+      wallet.address,
+      SignerType.Default
+    )
+    const signatureBuffer = Buffer.concat([
+      ethUtils.toBuffer(signature).slice(0, 65),
+      ethUtils.toBuffer(SignatureType.Wallet),
+    ])
+    signature = '0x' + signatureBuffer.toString('hex')
+    return signature
+  } else if (walletType === WalletType.ERC1271) {
+    // | 32 byte | 32 byte |1 byte| 1 bytes |
+    // +---------|---------|------|---------+
+    // |    R    |    S    |  V   | type(5) |
+    // +---------|---------|------|---------+
+    let signature = ecSignOrderHash(
```

Comment:
`ecSignOrderHash` seems outdated and could possibly result in error because the signature returned could be in either RSV form or VRS form. Maybe use `signMessage`?

Reply:
revised

Reply:
The `signByMMPSigner` part looks good to me.